### ISSUE TITLE
config_example.ini, channels.py exception and EPG import issues

### ISSUE THREAD

**Issue by cookieisland** [rocky4546/tvheadend-locast#9]
Hi Rocky4546

I tried your latest and found a couple of issues. I would have submitted a PR but I messed up my tvheadend configuration so badly I restored everything from last night's backup because I do not have more time today to work on this.  I had addressed 3 issues and was working on a 4th when I messed up and had to restore.  Here is what I ran into:

1) lib/tvheadend/config_example.ini should have 'login-' added to username and password like this:
```
[locast]
login-username = <locast username>
login-password = <locast password>
```

2) I get an exception in plugins/provider_video_locast/lib/channels.py.  on line 111 the variable 'hd' is used prior to it being initialized. I guess the first channel found in my listings did not have a 'videoProperties' so hd did not get set. I added a 'hd = 0' at the top of the 'for loop' and it cleared things up.

3) Due to the changes to the channels.m3u file tvheadend deleted all my old tvheadend muxes and added new ones with the new play urls plus other updates.  Thus I had to map all the new muxes to services and update all the channel icons that I set previously.  

4) For some reason the tvheadend EPG import didn't work correctly.  The epg seemed to import just fine when I ran tv_grab_url however the imported epg had programs for each channel but there were no titles for each entry.   The entries had icons and such but no title or description text.   I had previously used tv_grab_url from https://gist.github.com/saihtaM/5242283/raw/tv_grab_url.py and it worked fine with your previous versions but since you have provided a different one I figured I would try yours.  However that is when I messed up my tvheadend configuration and I ended up restoring both tvheadend and tvheadend-locast from my last night's backup.  

I have a theory that the new formatting of title and desc in xmltv.xml was causing the problem but since I had to restore from backup I can not work on this more today...

Example of a programme in the new xmltv.xml (I got no title and description text from this in tvheadend)

```
<programme start="20210426000000 +0000" stop="20210426003000 +0000" channel="1316">
		<title text="NBC Bay Area News at 5"/>
		<desc text="(2011/07/27) News / 
The latest local news, weather and investigative stories impacting the community."/>
		<video>
			<quality>HDTV</quality>
		</video>
		<date lang="en" text="20110727"/>
		<length units="minutes" text="30"/>
		<category lang="en" text="News / Current affairs"/>
		<icon src="https://fans.tmsimg.com/assets/p8776719_b_v5_ac.jpg"/>
	</programme>
```

The same programme in the old xmltv.xml (this worked for me)

```
<programme start="20210426000000 +0000" stop="20210426003000 +0000" channel="1316">
	<title lang="en">NBC Bay Area News at 5</title>
	<desc lang="en">(2011/07/27) News / 
The latest local news, weather and investigative stories impacting the community.</desc>
	<video>
		<quality>HDTV</quality>
	</video>
	<date lang="en">20110727</date>
	<length units="minutes">1800</length>
	<category lang="en">News / Current affairs</category>
	<genre lang="en">News / Current affairs</genre>
	<icon src="https://fans.tmsimg.com/assets/p8776719_b_v5_ac.jpg" />
	<rating><value>N/A</value></rating>
</programme>
```

**Comment by rocky4546** [comment]
I'll make the fixes and let you know what I found.  thanks for taking a look

**Comment by rocky4546** [comment]
1. Updated example.ini.  modded the two login lines and removed [player] section.
2. Why I don't get the error is surprising.  It is correct and it could not be set.  I must have something configured to not let me know this.  Added the hd = 0 at the top of the for loop
3. yep, tvheadend is notorious about doing that.  My solution is to have the channels.db a column called thumbnail which allows user editing.  That should help with the issue.  Just need to add a way to force an update to the database files when a new version is added. I keep removing the databases, but know the longer term method cannot be that.  The channel number, channel name and thumbnail are added fields for user input in the channels.db.
4. WOW, missed that one completely.  It looks pretty screwed up.  Will need to take a look.  I removed the lang='en' since some are other languages andthere is no idea what lang it really is.  Will probably just put it back in and let it happen.  Give me a little time to fix it.
